### DISCUSSION TITLE
Associated surfaces extend out beyond the site extent

### DISCUSSION THREAD

**Discussion by dbaumer21** [Riverscapes/gcd#424] [I need help please]
When deriving point density and slope associated surfaces, I get a raster that extends out of the site (a large square). I used a .shp file from the raw topographic data. I am also getting a message that says the projection doesn't match, and the authority doesn't match the coordinates. How do I fix that? I am guessing that might be the problem. Here are some pictures: 

![image](https://user-images.githubusercontent.com/98793929/152020199-3df35ad6-e112-4415-9a57-a426dba10330.png)

![image](https://user-images.githubusercontent.com/98793929/152020398-1fe8b21c-9853-4200-884b-49385ca5b96b.png)

Another unrelated question: What are the pros/cons of using a round vs. a square derivation method?

**Comment by philipbaileynar** [comment]
If you are convinced that the source ShapeFile has the correct projection then use the ArcGIS [Define Projection](https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/define-projection.htm) tool, using one of your DEMs as the source of the map projection. This will not alter the source ShapeFile data. It will merely redefine the map project associated with the ShapeFile.

If however there is the slightest chance that the source ShapeFile does indeed have a different map projection then you should use the ArcGIS [Project](https://pro.arcgis.com/en/pro-app/latest/tool-reference/data-management/project.htm) GeoProcessing tool, again using one of your DEMs as the target map projection, to reproject the source ShapeFile to the same map projection used by your DEMs. This will alter the source ShapeFile!

The links above are to ArcPro, but the same tools exist in ArcGIS 10.x.

**Comment by dbaumer21** [comment]
Thank you!

**Comment by joewheaton** [comment]
@philipbaileynar I have seen this behavior @dbaumer21 describes previously and I'm pretty sure we posted it in #351. I don't think this has anything to do with an incorrect projection in my experience. It has to do with the associated surfaces that we create for the user NOT effectively applying the NoData mask.  This is something that the ESRI geoprocessing tools do too incorrectly. I just don't think we're applying our own workflows and rules. I think these density algorithms are correctly applying a moving window and calculating zero for the rest of that data extent instead of using the NoData mask.

@dbaumer21 you asked:

> Another unrelated question: What are the pros/cons of using a round vs. a square derivation method?

The round tends to have a smoother transition and produce slightly lower point counts because it is sampling a smaller area for the same radius/cell count out. This is of little consequence and is really only something for the person making the fuzzy inference system to be aware of in what they use/take to mean low, medium and high. I would just stick with the square as it is what is more commonly used in literature when people report point density.

**Reply by philipbaileynar** [reply]
Are there two problems here? #351 is legit and needs addressing. But the projection warning feels like a separate issue?

**Reply by joewheaton** [reply]
You might be right @philipbaileynar. I am not sure about the projection thing, but I suspect it is separate issue.
